src/target/rp.c

```diff
@@ -408,7 +408,7 @@ bool rp_probe(target *t)
 	}
 	rp_add_flash(t, XIP_FLASH_START, size << 1);
 	t->driver = RP_ID;
-	target_add_ram(t, SRAM_START, 0x40000);
+	target_add_ram(t, SRAM_START, 0x42000);
 	target_add_ram(t, 0x51000000,  0x1000);
```

Comment:
By the way, is there a reason to not add the ranges of the peripherals here? It seems other targets also do not include the peripherals.

It would be nice for debugging if it were easily possible to read and write hardware registers through the debugger.

Reply:
It's mostly a matter of code size and not having included a way to define them correctly.. however, as you'll [find on the wiki](https://github.com/blackmagic-debug/blackmagic/wiki/Useful-GDB-commands#enabling-access-to-memory-mapped-io-from-gdb), you can work around this problem pretty trivially  till we can address it.